### ISSUE TITLE
Dumped core when building llvm and git packages

### ISSUE THREAD

**Issue by Izorkin** [NixOS/nixpkgs#139639]
**Describe the bug**
When building llvm and git packages, massive coredump errors are displayed in the journal

**Steps To Reproduce**
Run build llvm or git

Journal log:
```
...
kernel: show_signal: 85 callbacks suppressed
kernel: traps: git[122463] general protection fault ip:4eeb50 sp:7fffffff7ea0 error:0 in git[405000+25b000]
systemd[1]: Created slice system-systemd\x2dcoredump.slice.
systemd[1]: Started Process Core Dump (PID 122491/UID 0).
systemd-coredump[122648]: Resource limits disable core dumping for process 122463 (git).
systemd-coredump[122648]: [🡕] Process 122463 (git) of user 30011 dumped core.
systemd[1]: systemd-coredump@0-122491-0.service: Succeeded.
kernel: traps: git[123836] general protection fault ip:4eeb50 sp:7fffffff83b0 error:0 in git[405000+25b000]
systemd[1]: Started Process Core Dump (PID 123869/UID 0).
systemd-coredump[123980]: Resource limits disable core dumping for process 123836 (git).
systemd-coredump[123980]: [🡕] Process 123836 (git) of user 30011 dumped core.
systemd[1]: systemd-coredump@1-123869-0.service: Succeeded.
kernel: git[125117]: segfault at 40 ip 0000000000623e29 sp 00007fffffff7150 error 4 in git[405000+25b000]
kernel: Code: e0 00 00 00 4c 89 f6 83 c3 01 e8 d2 f4 ff ff 39 5c 24 10 0f 84 80 07 00 00 49 8b 96 58 02 00 00 48 63 c3 48 8b 12 48 8b 34 c2 <8b> 4e 40 48 89b4 24 e0 00 00 00 8d 41 6d 39 e8 0f 8e 71 ff ff ff
systemd[1]: Started Process Core Dump (PID 125147/UID 0).
systemd-coredump[125260]: Resource limits disable core dumping for process 125117 (git).
systemd-coredump[125260]: [🡕] Process 125117 (git) of user 30011 dumped core.
systemd[1]: systemd-coredump@2-125147-0.service: Succeeded.
kernel: git[127338]: segfault at 40 ip 0000000000623e29 sp 00007fffffff7340 error 4 in git[405000+25b000]
kernel: Code: e0 00 00 00 4c 89 f6 83 c3 01 e8 d2 f4 ff ff 39 5c 24 10 0f 84 80 07 00 00 49 8b 96 58 02 00 00 48 63 c3 48 8b 12 48 8b 34 c2 <8b> 4e 40 48 89b4 24 e0 00 00 00 8d 41 6d 39 e8 0f 8e 71 ff ff ff
systemd[1]: Started Process Core Dump (PID 127395/UID 0).
systemd-coredump[127515]: Resource limits disable core dumping for process 127338 (git).
systemd-coredump[127515]: [🡕] Process 127338 (git) of user 30011 dumped core.
systemd[1]: systemd-coredump@3-127395-0.service: Succeeded.
kernel: hrtimer: interrupt took 21011 ns
...
systemd[1]: Started Process Core Dump (PID 51698/UID 0).
systemd-coredump[51811]: Resource limits disable core dumping for process 51690 (CodeGenTests).
systemd-coredump[51811]: [🡕] Process 51690 (CodeGenTests) of user 30001 dumped core.
systemd[1]: systemd-coredump@4-51698-0.service: Succeeded.
systemd[1]: Started Process Core Dump (PID 55830/UID 0).
systemd-coredump[55848]: Resource limits disable core dumping for process 55800 (SupportTests).
systemd-coredump[55848]: [🡕] Process 55800 (SupportTests) of user 30001 dumped core.
systemd[1]: systemd-coredump@5-55830-0.service: Succeeded.
systemd[1]: Started Process Core Dump (PID 66605/UID 0).
systemd[1]: Started Process Core Dump (PID 66618/UID 0).
systemd[1]: Started Process Core Dump (PID 66841/UID 0).
systemd-coredump[66619]: Resource limits disable core dumping for process 66490 (llvm-as).
systemd-coredump[66619]: [🡕] Process 66490 (llvm-as) of user 30001 dumped core.
systemd[1]: systemd-coredump@6-66605-0.service: Succeeded.
systemd-coredump[66637]: Resource limits disable core dumping for process 66494 (llvm-as).
systemd-coredump[66637]: [🡕] Process 66494 (llvm-as) of user 30001 dumped core.
systemd[1]: systemd-coredump@7-66618-0.service: Succeeded.
systemd-coredump[66863]: Resource limits disable core dumping for process 66793 (opt).
systemd-coredump[66863]: [🡕] Process 66793 (opt) of user 30001 dumped core.
systemd[1]: systemd-coredump@8-66841-0.service: Succeeded.
systemd[1]: Started Process Core Dump (PID 67161/UID 0).
systemd[1]: Started Process Core Dump (PID 67163/UID 0).
systemd[1]: Started Process Core Dump (PID 67182/UID 0).
systemd[1]: Started Process Core Dump (PID 67181/UID 0).
systemd[1]: Started Process Core Dump (PID 67183/UID 0).
systemd[1]: Started Process Core Dump (PID 67184/UID 0).
systemd[1]: Started Process Core Dump (PID 67192/UID 0).
systemd[1]: Started Process Core Dump (PID 67187/UID 0).
systemd[1]: Started Process Core Dump (PID 67197/UID 0).
systemd[1]: Started Process Core Dump (PID 67204/UID 0).
kernel: Pid 63293(llvm-as) over core_pipe_limit
kernel: Skipping core dump
systemd[1]: Started Process Core Dump (PID 67196/UID 0).
kernel: Pid 63298(llvm-as) over core_pipe_limit
kernel: Skipping core dump
systemd[1]: Started Process Core Dump (PID 67216/UID 0).
kernel: Pid 63303(llvm-as) over core_pipe_limit
kernel: Pid 63301(llvm-as) over core_pipe_limit
kernel: Skipping core dump
kernel: Skipping core dump
kernel: Pid 63294(llvm-as) over core_pipe_limit
kernel: Skipping core dump
systemd[1]: Started Process Core Dump (PID 67228/UID 0).
systemd[1]: Started Process Core Dump (PID 67231/UID 0).
kernel: Pid 63297(llvm-as) over core_pipe_limit
kernel: Skipping core dump
systemd[1]: Started Process Core Dump (PID 67234/UID 0).
kernel: Pid 63302(llvm-as) over core_pipe_limit
kernel: Skipping core dump
systemd[1]: Started Process Core Dump (PID 67235/UID 0).
kernel: Pid 63318(llvm-as) over core_pipe_limit
kernel: Skipping core dump
kernel: Pid 63312(llvm-as) over core_pipe_limit
kernel: Skipping core dump
kernel: Pid 63320(llvm-as) over core_pipe_limit
kernel: Skipping core dump
kernel: Pid 63319(llvm-as) over core_pipe_limit
kernel: Skipping core dump
systemd-coredump[67217]: Resource limits disable core dumping for process 67153 (llvm-as).
systemd-coredump[67217]: [🡕] Process 67153 (llvm-as) of user 30001 dumped core.
systemd-coredump[67171]: Resource limits disable core dumping for process 67114 (llvm-as).
systemd-coredump[67171]: [🡕] Process 67114 (llvm-as) of user 30001 dumped core.
systemd[1]: systemd-coredump@12-67181-0.service: Succeeded.
systemd[1]: systemd-coredump@9-67161-0.service: Succeeded.
systemd-coredump[67195]: Resource limits disable core dumping for process 67100 (llvm-as).
systemd-coredump[67195]: [🡕] Process 67100 (llvm-as) of user 30001 dumped core.
systemd[1]: systemd-coredump@10-67163-0.service: Succeeded.
systemd-coredump[67221]: Resource limits disable core dumping for process 67130 (llvm-as).
systemd-coredump[67221]: [🡕] Process 67130 (llvm-as) of user 30001 dumped core.
systemd[1]: systemd-coredump@13-67183-0.service: Succeeded.
systemd-coredump[67246]: Resource limits disable core dumping for process 67165 (llvm-as).
systemd-coredump[67246]: [🡕] Process 67165 (llvm-as) of user 30001 dumped core.
systemd-coredump[67239]: Resource limits disable core dumping for process 67164 (llvm-as).
systemd-coredump[67239]: [🡕] Process 67164 (llvm-as) of user 30001 dumped core.
systemd-coredump[67223]: Resource limits disable core dumping for process 67135 (llvm-as).
systemd-coredump[67223]: [🡕] Process 67135 (llvm-as) of user 30001 dumped core.
systemd-coredump[67278]: Resource limits disable core dumping for process 67180 (llvm-as).
systemd-coredump[67278]: [🡕] Process 67180 (llvm-as) of user 30001 dumped core.
systemd[1]: systemd-coredump@20-67216-0.service: Succeeded.
systemd[1]: systemd-coredump@17-67197-0.service: Succeeded.
systemd[1]: systemd-coredump@14-67184-0.service: Succeeded.
systemd[1]: systemd-coredump@24-67235-0.service: Succeeded.
...
```

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.19-hardened1, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.11pre318367.37280c84c3e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

Checked on branch e3e5fc9bf1c7cbd32a28e2c83a295c12a36e250d and 955d01c675cd8118404dd21b6fa50330520ef810

**Comment by Artturin** [comment]
same
```

 systemd-coredump[3988705]: [🡕] Process 3988598 (llc) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1965-3988673-0.service: Deactivated successfully.
 systemd[1]: systemd-coredump@1970-3988707-0.service: Deactivated successfully.
 systemd-coredump[3988728]: Resource limits disable core dumping for process 3988674 (llc).
 systemd-coredump[3988728]: [🡕] Process 3988674 (llc) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1969-3988711-0.service: Deactivated successfully.
 systemd-coredump[3988701]: Resource limits disable core dumping for process 3988585 (llc).
 systemd-coredump[3988754]: Resource limits disable core dumping for process 3988689 (llc).
 systemd-coredump[3988701]: [🡕] Process 3988585 (llc) of user 30012 dumped core.
 systemd-coredump[3988754]: [🡕] Process 3988689 (llc) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1964-3988672-0.service: Deactivated successfully.
 systemd[1]: systemd-coredump@1975-3988721-0.service: Deactivated successfully.
 systemd-coredump[3988750]: Resource limits disable core dumping for process 3988658 (llc).
 systemd-coredump[3988750]: [🡕] Process 3988658 (llc) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1974-3988720-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3989358/UID 0).
 systemd-coredump[3989451]: Resource limits disable core dumping for process 3989343 (llvm-re
 systemd-coredump[3989451]: [🡕] Process 3989343 (llvm-readobj) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1976-3989358-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3989943/UID 0).
 systemd[1]: Started Process Core Dump (PID 3989956/UID 0).
 systemd-coredump[3990024]: Resource limits disable core dumping for process 3989902 (llvm-ob
 systemd-coredump[3990024]: [🡕] Process 3989902 (llvm-objdump) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1977-3989943-0.service: Deactivated successfully.
 systemd-coredump[3990035]: Resource limits disable core dumping for process 3989907 (llvm-ob
 systemd-coredump[3990035]: [🡕] Process 3989907 (llvm-objdump) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1978-3989956-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3990869/UID 0).
 systemd-coredump[3990919]: Resource limits disable core dumping for process 3990821 (opt).
 systemd-coredump[3990919]: [🡕] Process 3990821 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1979-3990869-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3991768/UID 0).
 systemd-coredump[3991835]: Resource limits disable core dumping for process 3991736 (opt).
 systemd-coredump[3991835]: [🡕] Process 3991736 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1980-3991768-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3992484/UID 0).
 systemd-coredump[3992531]: Resource limits disable core dumping for process 3992392 (llvm-tb
 systemd-coredump[3992531]: [🡕] Process 3992392 (llvm-tblgen) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1981-3992484-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3997429/UID 0).
 systemd-coredump[3997514]: Resource limits disable core dumping for process 3997378 (opt).
 systemd-coredump[3997514]: [🡕] Process 3997378 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1982-3997429-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 3998520/UID 0).
 systemd[1]: Started Process Core Dump (PID 3998523/UID 0).
 systemd[1]: Started Process Core Dump (PID 3998525/UID 0).
 systemd-coredump[3998575]: Resource limits disable core dumping for process 3998503 (opt).
 systemd-coredump[3998575]: [🡕] Process 3998503 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1983-3998520-0.service: Deactivated successfully.
 systemd-coredump[3998585]: Resource limits disable core dumping for process 3998513 (opt).
 systemd-coredump[3998584]: Resource limits disable core dumping for process 3998504 (opt).
 systemd-coredump[3998584]: [🡕] Process 3998504 (opt) of user 30012 dumped core.
 systemd-coredump[3998585]: [🡕] Process 3998513 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1984-3998523-0.service: Deactivated successfully.
 systemd[1]: systemd-coredump@1985-3998525-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4000912/UID 0).
 systemd-coredump[4000963]: Resource limits disable core dumping for process 4000857 (llvm-lt
 systemd-coredump[4000963]: [🡕] Process 4000857 (llvm-lto2) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1986-4000912-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4001089/UID 0).
 systemd-coredump[4001109]: Resource limits disable core dumping for process 4001046 (opt).
 systemd-coredump[4001109]: [🡕] Process 4001046 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1987-4001089-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4001397/UID 0).
 systemd-coredump[4001492]: Resource limits disable core dumping for process 4001352 (opt).
 systemd-coredump[4001492]: [🡕] Process 4001352 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1988-4001397-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4012190/UID 0).
 systemd-coredump[4012261]: Resource limits disable core dumping for process 4012158 (opt).
 systemd-coredump[4012261]: [🡕] Process 4012158 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1989-4012190-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4022640/UID 0).
 systemd-coredump[4022698]: Resource limits disable core dumping for process 4022405 (opt).
 systemd-coredump[4022698]: [🡕] Process 4022405 (opt) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1990-4022640-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4034184/UID 0).
 systemd-coredump[4034259]: Resource limits disable core dumping for process 4034142 (llvm-lt
 systemd-coredump[4034259]: [🡕] Process 4034142 (llvm-lto2) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1991-4034184-0.service: Deactivated successfully.
 systemd[1]: Started Process Core Dump (PID 4034393/UID 0).
 systemd-coredump[4034424]: Resource limits disable core dumping for process 4034354 (llvm-lt
 systemd-coredump[4034424]: [🡕] Process 4034354 (llvm-lto2) of user 30012 dumped core.
 systemd[1]: systemd-coredump@1992-4034393-0.service: Deactivated successfully.
 ```
 
 output of `coredumpctl list`
 https://gist.github.com/Artturin/4b8d1da834c860d385bebfbb78c00289

**Comment by Artturin** [comment]
Found https://github.com/NixOS/nixpkgs/issues/154938 and https://github.com/NixOS/nixpkgs/issues/157757

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/issues/154938#issuecomment-1018825938

> TL;DR: Just a noisy failure of a feature test, which shouldn't impact anything.

closing
